lib/apus/adapter.ex

```diff
@@ -1,7 +1,12 @@
 defmodule Apus.Adapter do
   @moduledoc """
+  An adapter provides a way to communicate with a third-party SMS service
+  using a consistent API.
   """
 
-  @callback deliver(%Apus.Message{}, %{}) :: any
-  @callback handle_config(map) :: map
+  @doc "Send the SMS message."
+  @callback deliver(%Apus.Message{}, config :: map) :: {:ok, Apus.Message.t()} | {:error, any}
+
+  @doc "Initialize adapter configuration."
+  @callback handle_config(config :: map) :: config :: map
```

Comment:
Possible typo on the return type here?

I think this should be `:: map` instead of `:: config :: map`?